src/Compiler/Driver/GraphChecking/GraphProcessing.fs

```diff
@@ -180,7 +180,8 @@ let processGraphAsync<'Item, 'Result when 'Item: equality and 'Item: comparison>
         let! parentCt = Async.CancellationToken
         use localCts = new CancellationTokenSource()
 
-        let completionSignal = TaskCompletionSource()
+        let completionSignal =
+            TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously)
```

Comment:
Why has this changed?

Reply:
This probably should be restored. It's a remnant of fighting what looked to me like odd deadlocks, and [this blogpost](https://devblogs.microsoft.com/premier-developer/the-danger-of-taskcompletionsourcet-class/) inspired me to try it.